### PR TITLE
Implement HW accelerated CRC

### LINKED ISSUE #25: HW accelerated CRC

### PR DESCRIPTION
**Pull request by MateuszMyalski**
Closes #25

### CODE REVIEW

File: middleware/libs/crc/src/crc32.c [hunk 1]
```diff
@@ -1,43 +1,95 @@
...
+}
+
+uint32_t crc32(const crc32_ctx_t *ctx, const uint8_t *data_in, size_t size) {
+    if ((NULL == ctx) || (NULL == data_in) || (0 > size)) {
+        return 0xFFFFFFFF;
+    }
+
+    uint32_t crc = ctx->init_crc;
+
+    /* Calculate how many even words contain the array (floor the division) */
```
Comment (bgs94) [inline middleware/libs/crc/src/crc32.c:74]: Should be "... the array contains ..."?

**Review by bgs94 — APPROVED** [review]
Just a bunch of cosmetic issues spotted. Merge it after resolving.

File: middleware/libs/crc/inc/crc32.h [hunk 2]
```diff
@@ -1,16 +1,27 @@
 #ifndef CRC32_H
 #define CRC32_H
 
-#include <stdint.h>
 #include <stddef.h>
+#include <stdint.h>
+#include <stdbool.h>
+
+typedef enum { CRC32_IN_NO_REVERSE, CRC32_IN_BYTE_REVERSE } crc32_in_reversed_t;
```
[the remark this answers is not in the document]
Comment (MateuszMyalski) [reply middleware/libs/crc/inc/crc32.h:8]: Unfortunately  or CLang format does not suppert feature to split enums this short to new lines. We can force to upgrade the clang -format version or leave it as is

[the remark this answers is not in the document]
Comment (bgs94) [reply middleware/libs/crc/inc/crc32.h:8]: Ok, let's leave it as it is for now.

**Comment by MateuszMyalski** [comment]
@bgs94 Resolved comments. Please merge :)

**Pull request merged**